lib/index.js

```diff
@@ -92,10 +110,85 @@ class History extends new Record(DEFAULTS) {
         if (!this.canRedo) return this;
 
         return this.merge({
-            undos: this.undos.push(current),
-            redos: this.redos.pop()
+            undos: this.undos.push(snapshot(current, this.merged)),
+            redos: this.redos.pop(),
+            merged: this.next.merged
         });
     }
+
+    /**
+     * Prune undo/redo using the defined strategy,
+     * after pushing a value on a valid History.
+     * @return {History}
+     */
+    prune() {
+        if (this.undos.size <= this.maxUndos) {
+            return this;
+        }
+
+        switch (this.strategy) {
```

Comment:
Maybe `this.strategy` can be a function so that other strategies can be implemented.